af/oometa/013-ooexplr-meta.install.sql

```diff
@@ -0,0 +1,81 @@
+BEGIN;
+
+select _v.register_patch( '013-ooexplr-meta', ARRAY[ '012-sha256-input-uniq' ], NULL );
+
+CREATE TABLE ooexpl_recent_msmt_count AS
+SELECT
+COUNT(msm_no) as msmt_count,
+probe_cc,
+probe_asn,
+test_name,
+test_start_time,
+bucket_date
+FROM measurement
+JOIN report ON report.report_no = measurement.report_no
+JOIN autoclaved ON autoclaved.autoclaved_no = report.autoclaved_no
+WHERE test_start_time > current_date - interval '31 day'
+GROUP BY probe_cc, probe_asn, test_name, test_start_time, bucket_date;
+
+ALTER TABLE ooexpl_recent_msmt_count
+ADD PRIMARY KEY(probe_asn, probe_cc, test_name, test_start_time, bucket_date);
+
+CREATE INDEX "ooexpl_recent_msmt_count_probe_cc_idx" ON "public"."ooexpl_bucket_msmt_count"("probe_cc");
+CREATE INDEX "ooexpl_recent_msmt_count_test_name_idx" ON "public"."ooexpl_bucket_msmt_count"("test_name");
+CREATE INDEX "ooexpl_recent_msmt_count_test_start_time_idx" ON "public"."ooexpl_bucket_msmt_count"("test_start_time");
+
+CREATE TABLE ooexpl_bucket_msmt_count AS
+SELECT
+COUNT(msm_no) as count,
+probe_asn,
+probe_cc,
+bucket_date
+FROM measurement 
+JOIN report ON report.report_no = measurement.report_no
+JOIN autoclaved ON autoclaved.autoclaved_no = report.autoclaved_no
+GROUP BY bucket_date, probe_asn, probe_cc;
+
+ALTER TABLE ooexpl_bucket_msmt_count
+ADD PRIMARY KEY(probe_asn, probe_cc, bucket_date);
+
+CREATE TABLE ooexpl_website_msmts AS
+SELECT
+measurement.msm_no,
+input.input,
+probe_asn,
+probe_cc,
+anomaly,
+confirmed,
+msm_failure as failure,
+blocking,
+http_experiment_failure,
+dns_experiment_failure,
+control_failure,
+bucket_date
+FROM measurement
+JOIN input ON input.input_no = measurement.input_no 
+JOIN report ON report.report_no = measurement.report_no
+JOIN http_verdict ON http_verdict.msm_no = measurement.msm_no
+JOIN autoclaved ON autoclaved.autoclaved_no = report.autoclaved_no
+WHERE test_name = 'web_connectivity' AND measurement_start_time > current_date - interval '31 day';
```

Comment:
It's a bit unclear, what's the long-term goal here. Is the goal to provide a 30d sliding window? Seems, it's not as "recent" is not the word that is used for the table name here. Is it a temporary ad-hoc workaround for ooni/api#59?

Reply:
Yes exactly. For the purpose of OONI Explorer we are going to in the initial MVP phase support showing the latest 30 days worth of measurements, but we may expand this to a larger window (though probably not larger than say 6 months), hence the name "recent".

If ooni/api#59 gets fixed it may make sense to drop this table, though maybe there is some performance tuning that we can do easier or better with a dedicated table?